crates/common/src/compile.rs

```diff
@@ -457,34 +458,50 @@ pub struct ContractInfo {
     pub is_dev_contract: bool,
 }
 
+/// Finds the path of the contract with the given name.
+/// Throws error if multiple or no contracts with the same name are found.
+pub fn find_contract_path(target_name: &str, project: &Project) -> Result<PathBuf> {
+    let graph = Graph::resolve(&project.paths)?;
+    let mut target = None;
+
+    for file in graph.files().keys() {
+        let src = std::fs::read_to_string(file)?;
+        let (parsed, _) = solang_parser::parse(&src, 0)
+            .map_err(|_| eyre::eyre!("Failed to parse {}.", file.display()))?;
```

Comment:
yeah I think we can do that

actually thought that we assumed solang being able to parse any valid Solidity

perhaps its better to invoke solc with empty output instead? similar to how we do it for tests now requesting just abi

Reply:
> actually thought that we assumed solang being able to parse any valid Solidity

it could be that future lang features are not supported right away, so I think we can fallback to previous behaviour if parsing fails.
if there's any syntax error this should also ensure that the user gets the proper solc error message

Reply:
updated code, now we fallback to running solc with empty output when solang fails

not really want to fully fallback to old behavior as this is not the most efficient way and will add additional complexity because we'd have to handle potential `find_contract_path` errors everywhere